source/slang/slang-lower-to-ir.cpp

```diff
@@ -6340,7 +6354,11 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
             SLANG_UNREACHABLE("associatedtype should have been handled by visitAssocTypeDecl.");
         }
 
-        bool isPublicType = decl->findModifier<PublicModifier>() != nullptr;
+        // TODO(JS): 
+        // Not clear what to do around HLSLExportModifier. 
+        // The HLSL spec says it only applies to functions, so we ignore for now.
+
+        const bool isPublicType = decl->findModifier<PublicModifier>() != nullptr;
```

Comment:
I  think the long/short here is that the presence/absence of `export` shouldn't affect whether something is considered "public" from the Slang perspective *at all* (at least that's a reasonable starting point...).

There's a bunch of messy stuff confounding the discussion here because "public" is currently getting used to mean a few different things inside of the Slang codebase, and the current `public` keyword, `PublicModifier`, and `PublicDecoration` all mean something pretty much categorically *wrong*.